src/modules/variable/init.rs

```diff
@@ -32,47 +40,144 @@ impl SyntaxModule<ParserMetadata> for VariableInit {
 
     fn new() -> Self {
         VariableInit {
-            name: String::new(),
+            definitions: vec![],
             expr: Box::new(Expr::new()),
-            global_id: None,
+            is_global_ctx: false,
             is_fun_ctx: false,
-            is_const: false
+            is_const: false,
         }
     }
 
     fn parse(&mut self, meta: &mut ParserMetadata) -> SyntaxResult {
         let keyword = token_by(meta, |word| ["let", "const"].contains(&word.as_str()))?;
         self.is_const = keyword == "const";
-        // Get the variable name
-        let tok = meta.get_current_token();
-        self.name = variable(meta, variable_name_extensions())?;
-        context!({
-            token(meta, "=")?;
-            syntax(meta, &mut *self.expr)?;
-            // Add a variable to the memory
-            self.handle_add_variable(meta, tok)?;
-            self.is_fun_ctx = meta.context.is_fun_ctx;
-            Ok(())
-        }, |position| {
-            error_pos!(meta, position, format!("Expected '=' after variable name '{}'", self.name))
-        })
+
+        let mut definitions: Vec<VariableDefinition> = vec![];
+        let mut is_destructured = false;
+
+        match token(meta, "[") {
+            Ok(_) => {
+                is_destructured = true;
+                let mut idx = 0;
+                loop {
+                    if token(meta, "]").is_ok() {
+                        break;
+                    }
+                    if idx > 0 {
+                        token(meta, ",")?;
+                    }
+                    let tok = meta.get_current_token();
+                    let name = variable(meta, variable_name_extensions())?;
+                    definitions.push(VariableDefinition {
+                        name: name.clone(),
+                        tok,
+                        global_id: None,
+                    });
+                    idx += 1;
+                }
+            }
+            Err(_) => {
+                let tok = meta.get_current_token();
+                let name = variable(meta, variable_name_extensions())?;
+                definitions.push(VariableDefinition {
+                    name: name.clone(),
+                    tok,
+                    global_id: None,
+                });
+            }
+        }
+
+        context!(
+            {
+                token(meta, "=")?;
+                syntax(meta, &mut *self.expr)?;
+
+                if is_destructured && !self.expr.get_type().is_array() {
+                    panic!("Expected array type for destructured variable");
```

Comment:
obviously a panic is bad, I am not yet familiar with the existing codebase, so this is temporary